lib/ansible/modules/identity/keycloak/keycloak_realm.py

```
@@ -0,0 +1,1063 @@
+        admin_theme=dict(type='str', aliases=['adminTheme']),
+        attributes=dict(type='dict'),
+        browser_flow=dict(type='str', aliases=['browserFlow']),
+        browser_security_headers=dict(type='dict', aliases=['browserSecurityHeaders'], options=dict(
+            contentSecurityPolicy=dict(type='str'),
+            xContentTypeOptions=dict(type='str'),
+            xFrameOptions=dict(type='str'),
+            xRobotsTag=dict(type='str'),
+            xXSSProtection=dict(type='str'),
+            strictTransportSecurity=dict(type='str'),
+        )),
+        brute_force_protected=dict(type='bool', aliases=['bruteForceProtected']),
+        client_authentication_flow=dict(type='str', aliases=['clientAuthenticationFlow']),
+        default_groups=dict(type='list', aliases=['defaultGroups']),
+        default_locale=dict(type='str', aliases=['defaultLocale']),
+        default_roles=dict(type='list', elements='str', aliases=['defaultRoles']),
+        direct_grant_flow=dict(type='str', aliases=['directGrantFlow']),
+        display_name=dict(type='str', aliases=['displayName']),
+        display_name_html=dict(type='str', aliases=['displayNameHtml']),
+        docker_authentication_flow=dict(type='str', aliases=['dockerAuthenticationFlow']),
+        duplicate_emails_allowed=dict(type='bool', aliases=['duplicateEmailsAllowed']),
+        edit_username_allowed=dict(type='bool', aliases=['editUsernameAllowed']),
+        email_theme=dict(type='str', aliases=['emailTheme']),
+        enabled=dict(type='bool'),
+        enabled_event_types=dict(type='list', elements='str', aliases=['enabledEventTypes']),
+        events_enabled=dict(type='bool', aliases=['eventsEnabled']),
+        events_expiration=dict(type='int', aliases=['eventsExpiration']),
+        events_listeners=dict(type='list', elements='str', aliases=['eventsListeners']),
+        failure_factor=dict(type='int', aliases=['failureFactor']),
+        id=dict(type='str'),
```

The module mentions that bugs appear if this is not the same as realm. Should it be exposed to the user?

---

Probably yes; there may or may not be a case where it is needed, and the API does expose it. I tried exposing the entirety of the API as Keycloak presents it; it is definitely possible to do "impossible" or "wrong" things with the API as exposed creating logically inconsistent states. Some can be caught in the module, but I do not have insight into how Keycloak does everything internally.
The reason I mentioned it in the docs is that when trying to feed it arbitrary data I ran into some weird errors, and that one might not want to repeat that experiment.

On the other hand, Keycloak is pretty extensible and I can't say for certain that this functionality is not being used by something out there.